src/Compilers/Core/Portable/InternalUtilities/ReferenceEqualityComparer.cs

```diff
@@ -23,6 +23,11 @@ bool IEqualityComparer<object>.Equals(object a, object b)
 
         int IEqualityComparer<object>.GetHashCode(object a)
         {
+            return ReferenceEqualityComparer.GetHashCode(a);
+        }
+
+        public static int GetHashCode(object a)
```

Comment:
I don't think it makes sense for callers to use this instead of `RuntimeHelpers.GetHashCode` directly.